src/Microsoft.AspNet.Mvc.Razor/TagHelpers/HiddenFormTagHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Mvc.Razor.TagHelpers
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    [HtmlTargetElement("form")]
+    public class HiddenFormTagHelper : TagHelper
+    {
+        [HtmlAttributeNotBound]
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
+
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            // Push the new FormContext.
+            ViewContext.FormContext = new FormContext();
+
+            await context.GetChildContentAsync();
+
+            var formContext = ViewContext.FormContext;
+            if (formContext.HasFormData && formContext.FormData.ContainsKey("hiddenForCheckBoxList"))
```

Comment:
Use a constant key that's visible to this class, `MvcForm`, and `DefaultHtmlGenerator`. Suggest `nameof(DefaultHtmlGenerator.GenerateHiddenForCheckbox)` -- though I suggest (separately) that method should be renamed.